ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/TxStatsUnitTest.java

```diff
@@ -31,34 +28,37 @@ public void test() throws Exception
             
             A.begin();
             B.begin();
+
+            Assertions.assertEquals(2, TxStats.getInstance().getNumberOfInflightTransactions());
             
             B.commit();
             A.commit();
+
+            Assertions.assertEquals(0, TxStats.getInstance().getNumberOfInflightTransactions());
         }
         
         for (int i = 0; i < 100; i++)
         {
             AtomicAction A = new AtomicAction();
             
             A.begin();
+
+            Assertions.assertEquals(1, TxStats.getInstance().getNumberOfInflightTransactions());
             
             A.abort();
+
+            Assertions.assertEquals(0, TxStats.getInstance().getNumberOfInflightTransactions());
         }
-        
-        AtomicAction B = new AtomicAction();
-        
-        B.begin();
-        
-        assertTrue(TxStats.enabled());
-        assertEquals(100, TxStats.getInstance().getNumberOfAbortedTransactions());
-        assertEquals(100, TxStats.getInstance().getNumberOfApplicationRollbacks());
-        assertEquals(200, TxStats.getInstance().getNumberOfCommittedTransactions());
-        assertEquals(0, TxStats.getInstance().getNumberOfHeuristics());
-        assertEquals(1, TxStats.getInstance().getNumberOfInflightTransactions());
-        assertEquals(100, TxStats.getInstance().getNumberOfNestedTransactions());
-        assertEquals(0, TxStats.getInstance().getNumberOfResourceRollbacks());
-        assertEquals(0, TxStats.getInstance().getNumberOfTimedOutTransactions());
-        assertEquals(301, TxStats.getInstance().getNumberOfTransactions());
+
+        Assertions.assertTrue(TxStats.enabled());
```

Comment:
This previous assertions are using the stats so the statement either needs removing or moved to the start of the test or moved into an @Before method.

Reply:
Fair point. I'll move this in a @before method. Thanks Michael